Contribution guidelines: type checks in lists (#56761)
tgstation/tgstation: .github/CONTRIBUTING.md

	getter_turned_into_variable = condition ? VALUE_C : VALUE_D
```

### Avoid unnecessary type checks and obscuring nulls in lists

> Typecasting in `for` loops carries and implied `istype()` check that filters non-matching types, nulls included. The `as anything` key can be used to skip the check.

Reviewer: ```suggestion
Typecasting in `for` loops carries an implied `istype()` check that filters non-matching types, nulls included. The `as anything` key can be used to skip the check.
```

If we know the list is supposed to only contain the desired type then we want to skip the check not only for the small optimization it offers, but also to catch any null entries that may creep into the list.

> Nulls in lists tend to point to improperly-handled references, which cause hard deletes are are hard to debug. Generating a runtime on those cases is more often than not positive.

Reviewer: ```suggestion
Nulls in lists tend to point to improperly-handled references, making hard deletes hard to debug. Generating a runtime in those cases is more often than not positive.
```
